namespace/consts.go

```diff
@@ -70,14 +70,12 @@ var (
 
 func primaryReservedNamespace(lastByte byte) Namespace {
 	return Namespace{
-		Version: NamespaceVersionZero,
-		ID:      append(bytes.Repeat([]byte{0x00}, NamespaceIDSize-1), lastByte),
+		data: append([]byte{NamespaceVersionZero}, append(bytes.Repeat([]byte{0x00}, NamespaceIDSize-1), lastByte)...),
 	}
 }
 
 func secondaryReservedNamespace(lastByte byte) Namespace {
 	return Namespace{
-		Version: NamespaceVersionMax,
-		ID:      append(bytes.Repeat([]byte{0xFF}, NamespaceIDSize-1), lastByte),
+		data: append([]byte{NamespaceVersionMax}, append(bytes.Repeat([]byte{0xFF}, NamespaceIDSize-1), lastByte)...),
```

Comment:
There are different approaches to do the same within this PR. In another place it is done by:
```
data := []byte{version}
data = append(data, id...)
```
I personally find later to read easier.

It looks like local constructor without inner validation will be handy. It will remove the need to do append tricks every time and unify the construction:
`func newNamespace(v version, id []bytes)`

Reply:
Makes sense. I also have a few ideas of constructing the namespace from just the bytes instead of needing to pass them separately (similar to `From`)